pkg/mesh-networking/apply/applier.go

```
@@ -752,6 +755,63 @@ func getAppliedVirtualMesh(
 	return nil
 }
 
+func getAppliedEastWestIngressGateways(
+	virtualMeshes networkingv1.VirtualMeshSlice,
+	mesh *discoveryv1.Mesh,
+	destinations discoveryv1.DestinationSlice,
+) []*discoveryv1.MeshStatus_IngressGateway {
+	var selectedIngressGatewayList []*discoveryv1.MeshStatus_IngressGateway
+	for _, virtualMesh := range virtualMeshes {
+		for _, ingressGatewayServiceSelector := range virtualMesh.Spec.GetFederation().GetEastWestIngressGatewaySelectors() {
+			// Check if this selector applies to this mesh
+			var applies bool
+			for _, meshRef := range ingressGatewayServiceSelector.GetMeshes() {
+				if meshRef.GetName() == mesh.GetName() && meshRef.GetNamespace() == mesh.GetNamespace() {
+					applies = true
+					break
+				}
+			}
+			if !applies {
+				continue
+			}
+			for _, destination := range destinations {
+				// Currently, we can only create gateways out of kube service destination types.
+				if destination.Spec.GetExternalService() != nil {
+					continue
```

we'll want to throw an error here if the user selects a non-k8s destination, and propagate that to the status

---

We can just check it in the config target validator and put it in the status directly there.